### ISSUE TITLE
Unclear how to turn off "listening for events" notification

### ISSUE THREAD

**Issue by ian-kelling** [vector-im/riot-android#1058]
I looked through all the options twice to figure out how to turn off this notification, then after googling it, I found issue #300, which explained it goes away when turning off "enable background sync". The "enable background sync" setting should clearly state that it has a persistent notification, and ideally, the notification itself should have text that matches with the setting that enables it, for example, instead of "listening for events", it should say "background sync", to correspond for the setting. Also, separate bug probably, but this kind of notification should be avoided, I run syncthing which does background syncing with no notification.

**Comment by licaon-kter** [comment]
Yes, please add an option to toggle this off.

Yes, some ROMs (eg. Huawei or Xiaomi) are too aggresive at killing the apps, but plenty others are not, exclude it in Doze and carry on.

Eg. [Conversations](https://github.com/siacs/Conversations/), I've tested it (without GCM!) with Sony 5.1.1/6.0.1 and Samsung 5.0.2/6.0.1 and it always worked great without the need to force the foreground service (and the obligatory permanent notification).

**Comment by akien-mga** [comment]
I would also like having an option to hide this notification (while keeping all other event notifications).

**Comment by flocke** [comment]
Yeah, I would love an option to disable the permanent notification. If you exclude Riot from Battery optimisations it should not be killed so easily. As @licaon-kter already mentioned that is how Conversations does it and I had never any problems with it. Maybe also do what Conversations does and automatically prompt the user to disable Battery optimisations for Riot on first launch.

**Comment by hrj** [comment]
One more option could be to stop and start the service for every sync. This should hide the notification for most of the time and when visible it would convey some meaning to the user.

Since the default sync interval is 600 seconds and a typical sync request might last for about 6 seconds, the notification will be visible only for 1% of the time.

**Comment by jurf** [comment]
Yes, please add an option for this, it’s really annoying.

**Comment by flocke** [comment]
On Android O you can disable the notification channel for that notification.
Go to your phone Settings -> Apps & notifications -> Riot.im -> App notifications and disable the "Listen for events" category. That way you still get every other notification from Riot but the persistent "Listening for events" is disabled.
I additionally excluded Riot from battery optimizations to make sure it is not killed. I have not yet missed any messages.

**Comment by jurf** [comment]
@flocke unfortunately still Android N.

**Comment by dkanada** [comment]
Here is some information on the options in Conversations. When the app is first installed, the user does not recieve any prompts, but shortly after installing, I got the following dialog.

![screenshot_20180728-033306](https://user-images.githubusercontent.com/21353219/43341110-7486e5a2-9219-11e8-8fff-e45e0fbaa95a.jpg)

That was followed by another dialog which I assume is from the operating system instead of the app itself.

![screenshot_20180728-033254](https://user-images.githubusercontent.com/21353219/43341151-8c580cf6-9219-11e8-98b2-fc60326c1935.jpg)

In the settings, conversation has an option to enable a foreground service seen below.

![screenshot_20180728-034028](https://user-images.githubusercontent.com/21353219/43341170-a01664c2-9219-11e8-8194-19325c0e4d04.png)

The notification does not show up on the bar normally, but when the notification bar is expanded you can see the Conversations service notification.

![screenshot_20180728-033316](https://user-images.githubusercontent.com/21353219/43341253-d9f87914-9219-11e8-8900-060bd16f3d76.jpg)

![screenshot_20180728-034022](https://user-images.githubusercontent.com/21353219/43341262-dfaa48d8-9219-11e8-9cfe-0566f0a8ae18.png)

Conversations works perfectly fine with either the foreground service or the battery optimizations disabled. Personally I don't mind the foreground notification, but I have talked to a lot of other people who don't like it.

It looks like the android client already has code for [battery optimizations](https://github.com/vector-im/riot-android/blob/f01f805ce4f7f5d5d603ace32e0bb32af69e79a9/vector/src/main/java/im/vector/util/PreferencesManager.java#L222-L262) in the preference manager. Disabling battery optimizations enables background sync without a foreground process for many users on Conversations, so it should work here as well. Maybe the background sync setting could be modified to have two options (disabled battery optimizations or persistent notification) on newer devices using Marshmallow or later.

The two issues below are tangentially related to background sync so I wanted to include them in this issue thread.

#1801 #2196

**Comment by 1u** [comment]
I had the exact same problem. and eventually uninstalled riot.
@ all devs: I would fix most of the 323 [open bugs](https://github.com/vector-im/riot-android/labels/bug) before implementing any other improvement. 
Thx @ian-kelling for creating this issue and everyone contributing.
